File: android/app/src/main/java/com/djs66256/short_drama/domain/model/BookingAsset.kt
package com.djs66256.short_drama.domain.model

data class BookingAsset(
    val dramaId: String,
    val title: String,
    val coverUrl: String,
    val episodeCount: Int,
    val bookedAt: String,
    val availabilityStatus: BookingAssetStatus,
)

File: ios/ShortDrama/Sources/Domain/Entities/BookingAsset.swift
import Foundation

enum BookingAssetAvailabilityStatus: String, Codable, CaseIterable, Equatable, Sendable {
    case online
    case upcoming

    var title: String {
        switch self {
        case .online:
            return "已上线"
        case .upcoming:
            return "待上线"
        }
    }
}

struct BookingAsset: Identifiable, Equatable, Sendable {
    let dramaID: String
    let title: String
    let coverURL: String?
    let episodeCount: Int
    let bookedAt: String
    let availabilityStatus: BookingAssetAvailabilityStatus

    var id: String { dramaID }
}
